### PR TITLE
STY/TST: update with pysat 3.0.0 standards

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #3, #16 (partial), #27, #36 

- Streamline setup.py, move metadata to setup.cfg
- Use pysat testing methods where possible to reduce code duplication
- Use pysat instrument_test_class to handle instrument tests
- Use document tests in Travis CI
- Remove numeric `inst_id`, replace with `num_samples` and `freq`.
- Updates instrument standards

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update

# How Has This Been Tested?

```
import pysat
import pysatMissions
sgp4 = pysat.Instrument(inst_module=pysatMissions.instruments.pysat_sgp4, num_samples=100)
sgp4.load(2018, 1)
ephem = pysat.Instrument(inst_module=pysatMissions.instruments.pysat_ephem, num_samples=100)
ephem.load(2018, 1)
```
 `ephem.data` produces
```
                          glong       glat         alt  ...  sc_yhat_ecef_x  sc_yhat_ecef_y  sc_yhat_ecef_z
Epoch                                                   ...                                                
2018-01-01 00:00:00 -160.707179  14.739780  409.765719  ...             NaN             NaN             NaN
2018-01-01 00:00:01 -160.668439  14.690125  409.762625  ...       -0.407504        0.701192       -0.585038
2018-01-01 00:00:02 -160.629711  14.640463  409.759563  ...       -0.407504        0.701223       -0.585001
2018-01-01 00:00:03 -160.591012  14.590792  409.756500  ...       -0.407505        0.701255       -0.584962
2018-01-01 00:00:04 -160.552325  14.541114  409.753469  ...       -0.407504        0.701233       -0.584989
...                         ...        ...         ...  ...             ...             ...             ...
2018-01-01 00:01:35 -157.103790   9.991780  409.573719  ...       -0.407671        0.703100       -0.582627
2018-01-01 00:01:36 -157.066593   9.941515  409.572844  ...       -0.407675        0.703120       -0.582601
2018-01-01 00:01:37 -157.029396   9.891245  409.572000  ...       -0.407669        0.703091       -0.582640
2018-01-01 00:01:38 -156.992212   9.840970  409.571156  ...       -0.407673        0.703112       -0.582612
2018-01-01 00:01:39 -156.955042   9.790690  409.570344  ...             NaN             NaN             NaN

[100 rows x 27 columns]
```
`sgp4.data` produces
```
                     position_eci_x  position_eci_y  position_eci_z  velocity_eci_x  velocity_eci_y  velocity_eci_z
Epoch                                                                                                              
2018-01-01 00:00:00     3271.251386    -5689.980332     1726.226388        5.011837        1.149178       -5.686093
2018-01-01 00:00:01     3276.261110    -5688.827532     1720.539222        5.007658        1.156441       -5.688299
2018-01-01 00:00:02     3281.266651    -5687.667471     1714.849854        5.003473        1.163702       -5.690498
2018-01-01 00:00:03     3286.268005    -5686.500149     1709.158291        4.999281        1.170961       -5.692690
2018-01-01 00:00:04     3291.265364    -5685.325522     1703.464311        4.995083        1.178220       -5.694874
...                             ...             ...             ...             ...             ...             ...
2018-01-01 00:01:35     3727.637434    -5548.274156     1177.124744        4.587058        1.831253       -5.862810
2018-01-01 00:01:36     3732.222090    -5546.439376     1171.261214        4.582297        1.838334       -5.864313
2018-01-01 00:01:37     3736.801981    -5544.597516     1165.396185        4.577530        1.845413       -5.865809
2018-01-01 00:01:38     3741.377102    -5542.748578     1159.529665        4.572757        1.852489       -5.867297
2018-01-01 00:01:39     3745.947447    -5540.892566     1153.661660        4.567978        1.859564       -5.868777

[100 rows x 6 columns]
```

These outputs should be identical to running the develop branch using the `inst_id='100'` keyword for each instrument.

**Test Configuration**:
* Mac OS X 10.15.7
* Python 3.8.2

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
Added #40 to cover documentation updates required.  Otherwise, this is ready to go.

File: pysatMissions/instruments/_core.py [hunk 1]
```diff
@@ -2,43 +2,9 @@
...
-    # only in Mac. '%x' workins in both Windows and Mac
-    names = [data_path + date.strftime('%Y-%m-%d') + '.nofile'
-             for date in index]
-    return pds.Series(names, index=index)
-
-
-def _download(date_array, tag, inst_id, data_path=None):
-    """ Data is simulated so no download routine is possible. Simple pass
-    function"""
     pass
```
Comment (aburrell) [inline pysatMissions/instruments/_core.py:10]:
```suggestion
    return
```

File: pysatMissions/instruments/pysat_sgp4.py [hunk 2]
```diff
@@ -30,11 +30,14 @@ def init(self):
 
     """
 
+    self.acknowledgements = ''
+    self.references = ''
+
     pass
```
Comment (aburrell) [inline pysatMissions/instruments/pysat_sgp4.py:36]: Return, and also logging info.

File: pysatMissions/tests/test_instruments.py [hunk 3]
```diff
@@ -1,70 +1,68 @@
...
+# The following lines apply the custom instrument lists to each type of test
+method_list = [func for func in dir(InstTestClass)
+               if callable(getattr(InstTestClass, func))]
+# Search tests for iteration via pytestmark, update instrument list
+for method in method_list:
+    if hasattr(getattr(InstTestClass, method), 'pytestmark'):
+        # Get list of names of pytestmarks
+        Nargs = len(getattr(InstTestClass, method).pytestmark)
+        names = [getattr(InstTestClass, method).pytestmark[j].name
+                 for j in range(0, Nargs)]
```
Comment (aburrell) [inline pysatMissions/tests/test_instruments.py:32]: snake_case!

File: setup.cfg [hunk 4]
```diff
@@ -1,3 +1,39 @@
+[metadata]
+name = pysatMissions
+version = file: pysatMissions/version.txt
+url = https://github.com/pysat/pysatMissions
+author = Jeff Klenzing, Russell Stoneback
+author_email = pysat.developers@gmail.com
+description = 'Mission Planning toolkit for pysat'
+keywords =
+  pysat
+  ionosphere
```
Comment (aburrell) [inline setup.cfg:10]: moar?

File: .travis.yml [hunk 5]
```diff
@@ -34,21 +39,26 @@ install:
   # Useful for debugging any issues with conda
   - conda info -a
   # Create conda test environment
-  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION numpy scipy pandas xarray netCDF4 h5py pytest-cov pytest-ordering coveralls portalocker
+  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION
   - conda activate test-environment
+  # Once pysat 3.0 is released, remove the custom pysat install and change
+  # this to:
+  # - while read requirement; do conda install --yes $requirement; done < requirements.txt1
+  - conda install --yes scipy pandas xarray netCDF4 portalocker
```
Comment (aburrell) [inline .travis.yml:47]: improve through pysat requirements.

**Review by aburrell — APPROVED** [review]

**Pull request merged**